hvplot/tests/testgeo.py

```diff
@@ -140,6 +140,7 @@ def test_geo_with_rasterize(self):
             rasterize=True,
             dynamic=False,
             aggregator="max",
+            project=True,
```

Comment:
This looks good to me; however points is back to being broken without `project=True` which to me seems undesirable.

Can we keep the last PR's changes but with points?:
```python
        if self.geo and self.crs != self.output_projection and self.kind == "points":
            import geoviews as gv
            obj = gv.project(obj, projection=self.output_projection)
```

Reply:
Actually maybe putting it under the existing clause:
```python
        if self.crs and self.project or (self.kind == "points" and self.rasterize):
            # Apply projection before rasterizing
            import geoviews as gv
            obj = gv.project(obj, projection=self.output_projection)
```

Reply:
I would rather not. There is an extra cost associated with `gv.project`. It seems points work for the simple case, but we can't guarantee it will work for all cases. 

The documentation [here](https://hvplot.holoviz.org/user_guide/Geographic_Data.html#declaring-an-output-projection) could have some an [admonition](https://sphinx-book-theme.readthedocs.io/en/stable/reference/kitchen-sink/admonitions.html) that say try `project=True` if you see problem with projection and rasterize.

Reply:
> It seems points work for the simple case, but we can't guarantee it will work for all cases.

Do you mean if the points X*Y is large, it won't work?

Reply:
Or it will slow it significantly down with no way to disable it.

Reply:
Just tested:
We can test it with this:

With rasterize=True (no project), it takes 24s; with project, 40s
```python
import hvplot.xarray
import rioxarray as rxr

da = rxr.open_rasterio("https://oin-hotosm.s3.amazonaws.com/5d7dad0becaf880008a9bc88/0/5d7dad0becaf880008a9bc89.tif")
da.isel(band=0).hvplot.points(x="x", y="y", projection="Robinson")
```

I'll let you make the final call~